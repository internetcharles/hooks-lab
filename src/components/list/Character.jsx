import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image, occupation, status }) => (
  <figure>
    <h1>{name}</h1>
    <img src={image} alt={name} />
    <h3>{occupation}</h3>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired
};

export default Character;