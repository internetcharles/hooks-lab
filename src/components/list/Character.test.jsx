import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const { asFragment } = render(<Character 
      name="Phil"
      image="phil.jpg"
      occupation="Farmer"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});