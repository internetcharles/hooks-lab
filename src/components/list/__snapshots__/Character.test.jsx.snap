// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      Phil
    </h1>
    <img
      alt="Phil"
      src="phil.jpg"
    />
    <h3>
      Farmer
    </h3>
  </figure>
</DocumentFragment>
`;
